File: data/transcoder_evaluation_gfg/c/REMOVE_MINIMUM_ELEMENTS_EITHER_SIDE_2MIN_MAX.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

int f_gold ( int arr [ ], int n ) {
  int longest_start = - 1, longest_end = 0;
  for ( int start = 0;
  start < n;
  start ++ ) {
    int min = INT_MAX, max = INT_MIN;
    for ( int end = start;
    end < n;
    end ++ ) {
      int val = arr [ end ];
      if ( val < min ) min = val;
      if ( val > max ) max = val;
      if ( 2 * min <= max ) break;
      if ( end - start > longest_end - longest_start || longest_start == - 1 ) {
        longest_start = start;
        longest_end = end;
      }
    }
  }
  if ( longest_start == - 1 ) return n;
  return ( n - ( longest_end - longest_start + 1 ) );
}


int f_filled ( int arr [ ], int n ) {}

int main(void) {
    int n_success = 0;
    int param0_0[] = {32,50,66,73,76,87};
int param0_1[] = {68,74,16,40,6,-44,-36,94,6,-24,-4,-58,-16,24};
int param0_2[] = {0,0,0,0,0,0,0,0,0,1,1,1,1,1,1,1,1,1,1,1,1};
int param0_3[] = {91,22};
int param0_4[] = {-84,-80,-78,-76,-58,-54,-52,-48,-42,-42,-40,-38,-34,-32,-28,-24,-6,2,2,4,10,14,16,18,26,26,36,40,50,52,62,64,72,74,84,90,94,96,98};
int param0_5[] = {1,0,1,1,1,0,1,1,1,0,1,0,0,1,0,1,1,1,1,0,1};
int param0_6[] = {10,19,25,29,32,37,40,43,43,44,46,51,51,54,56,58,63,79,83,86,87,97,97};
int param0_7[] = {-48,-28,10,30,78,-72,78,52,-52,-68,56,42,8,-42,16,-56,2,-90,-26,-28,-56,-2,80,-50,98,-64,-96,10,-10,44,98,-48,-88,42,30,24,38,-26,-52,-12,0,34,-82,-80,0,-84,-20};
int param0_8[] = {0,0,1,1,1,1,1,1};
int param0_9[] = {25,82};
int *param0[10] = {param0_0,param0_1,param0_2,param0_3,param0_4,param0_5,param0_6,param0_7,param0_8,param0_9};
    int param1[] = {5,11,11,1,35,13,11,25,6,1};
    for(int i = 0; i < len(param0); ++i)
    {
        if(f_filled(param0[i],param1[i]) == f_gold(param0[i],param1[i]))
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}

File: data/transcoder_evaluation_gfg/rust/REMOVE_MINIMUM_ELEMENTS_EITHER_SIDE_2MIN_MAX.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut arr: *mut i32, mut n: i32)
 -> i32 {
    let mut longest_start: i32 = -(1 as i32);
    let mut longest_end: i32 = 0 as i32;
    let mut start: i32 = 0 as i32;
    while start < n {
        let mut min_0: i32 = 2147483647 as i32;
        let mut max_0: i32 =
            -(2147483647 as i32) - 1 as i32;
        let mut end: i32 = start;
        while end < n {
            let mut val: i32 = *arr.offset(end as isize);
            if val < min_0 { min_0 = val }
            if val > max_0 { max_0 = val }
            if 2 as i32 * min_0 <= max_0 { break ; }
            if end - start > longest_end - longest_start ||
                   longest_start == -(1 as i32) {
                longest_start = start;
                longest_end = end
            }
            end += 1
        }
        start += 1
    }
    if longest_start == -(1 as i32) { return n }
    return n - (longest_end - longest_start + 1 as i32);
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0_0: [i32; 6] =
        [32 as i32, 50 as i32, 66 as i32,
         73 as i32, 76 as i32, 87 as i32];
    let mut param0_1: [i32; 14] =
        [68 as i32, 74 as i32, 16 as i32,
         40 as i32, 6 as i32, -(44 as i32),
         -(36 as i32), 94 as i32, 6 as i32,
         -(24 as i32), -(4 as i32), -(58 as i32),
         -(16 as i32), 24 as i32];
    let mut param0_2: [i32; 21] =
        [0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         0 as i32, 0 as i32, 0 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32];
    let mut param0_3: [i32; 2] =
        [91 as i32, 22 as i32];
    let mut param0_4: [i32; 39] =
        [-(84 as i32), -(80 as i32), -(78 as i32),
         -(76 as i32), -(58 as i32), -(54 as i32),
         -(52 as i32), -(48 as i32), -(42 as i32),
         -(42 as i32), -(40 as i32), -(38 as i32),
         -(34 as i32), -(32 as i32), -(28 as i32),
         -(24 as i32), -(6 as i32), 2 as i32,
         2 as i32, 4 as i32, 10 as i32,
         14 as i32, 16 as i32, 18 as i32,
         26 as i32, 26 as i32, 36 as i32,
         40 as i32, 50 as i32, 52 as i32,
         62 as i32, 64 as i32, 72 as i32,
         74 as i32, 84 as i32, 90 as i32,
         94 as i32, 96 as i32, 98 as i32];
    let mut param0_5: [i32; 21] =
        [1 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 0 as i32,
         1 as i32, 1 as i32, 1 as i32,
         0 as i32, 1 as i32, 0 as i32,
         0 as i32, 1 as i32, 0 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 0 as i32, 1 as i32];
    let mut param0_6: [i32; 23] =
        [10 as i32, 19 as i32, 25 as i32,
         29 as i32, 32 as i32, 37 as i32,
         40 as i32, 43 as i32, 43 as i32,
         44 as i32, 46 as i32, 51 as i32,
         51 as i32, 54 as i32, 56 as i32,
         58 as i32, 63 as i32, 79 as i32,
         83 as i32, 86 as i32, 87 as i32,
         97 as i32, 97 as i32];
    let mut param0_7: [i32; 47] =
        [-(48 as i32), -(28 as i32), 10 as i32,
         30 as i32, 78 as i32, -(72 as i32),
         78 as i32, 52 as i32, -(52 as i32),
         -(68 as i32), 56 as i32, 42 as i32,
         8 as i32, -(42 as i32), 16 as i32,
         -(56 as i32), 2 as i32, -(90 as i32),
         -(26 as i32), -(28 as i32), -(56 as i32),
         -(2 as i32), 80 as i32, -(50 as i32),
         98 as i32, -(64 as i32), -(96 as i32),
         10 as i32, -(10 as i32), 44 as i32,
         98 as i32, -(48 as i32), -(88 as i32),
         42 as i32, 30 as i32, 24 as i32,
         38 as i32, -(26 as i32), -(52 as i32),
         -(12 as i32), 0 as i32, 34 as i32,
         -(82 as i32), -(80 as i32), 0 as i32,
         -(84 as i32), -(20 as i32)];
    let mut param0_8: [i32; 8] =
        [0 as i32, 0 as i32, 1 as i32,
         1 as i32, 1 as i32, 1 as i32,
         1 as i32, 1 as i32];
    let mut param0_9: [i32; 2] =
        [25 as i32, 82 as i32];
    let mut param0: [*mut i32; 10] =
        [param0_0.as_mut_ptr(), param0_1.as_mut_ptr(), param0_2.as_mut_ptr(),
         param0_3.as_mut_ptr(), param0_4.as_mut_ptr(), param0_5.as_mut_ptr(),
         param0_6.as_mut_ptr(), param0_7.as_mut_ptr(), param0_8.as_mut_ptr(),
         param0_9.as_mut_ptr()];
    let mut param1: [i32; 10] =
        [5 as i32, 11 as i32, 11 as i32,
         1 as i32, 35 as i32, 13 as i32,
         11 as i32, 25 as i32, 6 as i32,
         1 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize], param1[i as usize]) ==
               f_gold(param0[i as usize], param1[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
